src/main/docs/guide/kafkaQuickStart.adoc

```diff
@@ -71,7 +71,7 @@ At compile time Micronaut will produce an implementation of the above interface.
 
 snippet::io.micronaut.kafka.docs.quickstart.QuickstartTest[tags=quickstart, indent=0]
 
-Note that since the `sendProduct` method returns `void` this means the method will send the `ProducerRecord` and block until the response is received. You can return a `Future` or rs:Publisher[] to support non-blocking message delivery.
+Note that since the `sendProduct` method returns `void` this means the method will send the `ProducerRecord` and block until the response is received. You can specify an executor and return a `Future` or rs:Publisher[] to support non-blocking message delivery.
```

Comment:
Do you need to both specify an executor **and** return a `Future`?

Should this read 

> You can specify an executor, return a `Future`, or return a rs:Publisher...

?

Not sure :-)

Reply:
In short, yes.

The requirement I'm trying to convey is basically that `executor` is only applicable in conjunction with either a `Future` or `Publisher` return type, and if it's not used, they will still be blocking.

Maybe it should actually be:

"You can specify an executor and return either a `Future` or rs:Publisher[] to support non-blocking message delivery."

Does that make it more clear?

Reply:
Come to think of it, I think it also specifically needs to be a `CompletableFuture` and not just `Future`.

Reply:
> You can specify an executor and return either a `CompletableFuture` or rs:Publisher[] to support non-blocking message delivery.

👍